### PR TITLE
refactor!: remove items marked as deprecated in `0.2.1`

### PR DESCRIPTION
**Pull request by imrn99**
remove deprecated items and update the UG to mention utilities. Also, add the `utils` feature to the doc CI.

## Scope

- [x] Code: `honeycomb-core`

## Type of change

- [x] Refactor

## Other

- [x] Breaking change

### CODE REVIEW

File: honeycomb-guide/src/project-structure/honeycomb-core.md [hunk 1]
```diff
@@ -24,3 +25,8 @@ following:
 - **Orbit2**: Generic 2D implementation for orbit computation
 - **Vector2**: 2D vector representation
 - **Vertex2**: 2D vertex representation
+
+Note that optional features can be enabled when compiling this crate:
+
+- `utils` -- provides additionnal items that would qualify as quality-of-life, notably map generation functions.
```
Comment (cedricchevalier19) [inline honeycomb-guide/src/project-structure/honeycomb-core.md:31]: Not very clear from a user perspective.

File: honeycomb-guide/src/project-structure/honeycomb-core.md [hunk 2]
```diff
@@ -24,3 +25,8 @@ following:
...
+- `single_precision` -- uses `f32` instead of `f64` for coordinates representation in tests
```
Comment (cedricchevalier19) [inline honeycomb-guide/src/project-structure/honeycomb-core.md:32]: Do we need to expose that to the user?

**Review by cedricchevalier19 — APPROVED** [review]

File: honeycomb-guide/src/project-structure/honeycomb-core.md (hunk 2 above)
Comment (imrn99) [reply honeycomb-guide/src/project-structure/honeycomb-core.md:32]: Not really, I'm removing this mention from the UG

File: honeycomb-guide/src/project-structure/honeycomb-core.md (hunk 1 above)
Comment (imrn99) [reply honeycomb-guide/src/project-structure/honeycomb-core.md]: I rewrote this to be closer to the initial description of the feature

**Pull request merged**